rules/aip0231/method_rules.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0231
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/kjk/inflect"
+)
+
+var pluralMethodResourceName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0231", "method-name", "name"),
+	URI:    "https://aip.dev/231#guidance",
+	OnlyIf: isBatchGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `BatchGetFoos`
+		pluralMethodResourceName := m.GetName()[8:]
+
+		if !inflect.IsPlural(pluralMethodResourceName) {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					`The resource part in method name %q shouldn't %q, but should be its plural form %q`,
+					m.GetName(), pluralMethodResourceName, inflect.ToPlural(pluralMethodResourceName),
+				),
+				Descriptor: m,
+			}}
+		}
+
+		return nil
+	},
+}
+
+// Batch Get method should have a properly named Request message.
+var inputName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0231", "input-message", "name"),
+	URI:    "https://aip.dev/231#guidance",
+	OnlyIf: isBatchGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `BatchGetFoos`, the request
+		// message is named `BatchGetFoosRequest`.\
+		pluralInputResourceName := strings.Title(inflect.ToPlural(m.GetName()[8:]))
+
+		// Rule check: Establish that for methods such as `BatchGetFoos`, the request
+		// message is named `BatchGetFoosRequest`.
+		if got, want := m.GetInputType().GetName(), fmt.Sprintf("BatchGet%sRequest", pluralInputResourceName); got != want {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					"Batch Get RPCs should have a properly named request message %q, but not %q",
+					want, got,
+				),
+				Descriptor: m,
+			}}
+		}
+
+		return nil
+	},
+}
+
+// Batch Get method should have a properly named Response message.
+var outputName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0231", "output-message", "name"),
+	URI:    "https://aip.dev/231#guidance",
+	OnlyIf: isBatchGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `BatchGetFoos`, the response
```

Comment:
Can you remove the duplicated comments? Thanks!

Reply:
Done.